### DISCUSSION TITLE
Sharing devices to Amazon Web Services (AWS)

### DISCUSSION THREAD

**Discussion by d0n13** [dorssel/usbipd-win#405] [Show and tell]
Hi all

Trying to share a device to AWS linux install using usbipd and I'm wondering if anyone has got this to work over an SSH tunnel? I have tried to create the tunnel using the following but I'm not having any luck so far. 

```
ssh -i ./usbip-server.pem -Nf -R 3240:localhost:3240 ubuntu@ec2-xx.xx.xx.xx.eu-west-1.compute.amazonaws.com
```

Apart from the tunnel from being setup wrong will this even work? I know usbipd is already listening at port 3240 on the local machine so I wonder can I do something else to configure this to operate?

**Comment by dorssel** [comment]
@d0n13 
I've used this trick (ssh -R) to forward a USB device to a remote computer while the local USB host had a strict firewall not allowing any outside connection in. That worked fine. A single tunnel for 3240 should be enough.

Maybe the remote host is already using port 3240 itself (for its own usbipd, or possibly even something else). You could use a different remote port, for example: ssh -R 12345:localhost:3240. Then you should be able to use `usbip --tcp-port 12345 attach -r 127.0.0.1` on the remote end.

**Comment by d0n13** [comment]
Thanks @dorssel. I’ll give it a go. Thank you

**Comment by d0n13** [comment]
Am I also right in thinking that you don’t need the usbip daemon running on the aws side? Or does it need the full setup. Asking because it’s not straight forward to setup. Can I just use the usbip client on the aws side?

**Comment by dorssel** [comment]
@d0n13 
Correct, no need for usbipd on the client side, nor anywhere in the middle. Port forwarding is sufficient.

**Comment by d0n13** [comment]
Found a good guide here for Ubuntu 20.04 and it worked. 
https://askubuntu.com/questions/1303403/how-to-install-usbip-vhci-hcd-drivers-on-an-aws-ec2-ubuntu-kernel-version
Bit tricky to setup but the instructions are ok. Just remember you have to be root on the AWS side to execute the attach...

First created the SSH tunnel as follows:
```
 ssh -i .\ubuntu20.04.pem -R 2001:localhost:3240 ubuntu@ec2-xx-xx-xx-xx.eu-west-1.compute.amazonaws.com
```

Shared an iPhone on the Windows PC
```
PS C:\Users\donie> usbipd list
Connected:
BUSID  VID:PID    DEVICE                                                        STATE
1-10   27c6:533c  Goodix fingerprint                                            Not shared
1-11   0bda:5510  Integrated Webcam, Camera DFU Device                          Not shared
1-14   8087:0026  Intel(R) Wireless Bluetooth(R)                                Not shared
3-5    0bda:4014  Realtek USB Audio                                             Not shared
3-7    046d:c52b  Logitech USB Input Device, USB Input Device                   Not shared
4-2    0bda:8153  Realtek USB GbE Family Controller                             Not shared
6-1    05ac:12a8  Apple Mobile Device USB Composite Device                      Shared (forced)
```
Then attach on AWS side.

```
ubuntu@ip-xx-xx-xx-xx:~$ sudo usbip --tcp-port 2001 attach -r 127.0.0.1 -b 6-1
usbip: info: using port 2001 ("2001")
ubuntu@ip-172-31-35-88:~$ lsusb
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 002: ID 05ac:12a8 Apple, Inc. iPhone5/5C/5S/6
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
```

Installed libimobiledevice and was able to connect to the phone and retrieve details.
Thanks @dorssel for the tips. Were very helpful

**Comment by dorssel** [comment]
@d0n13 
Is it OK if I close this by converting it to a "show and tell" discussion?
And change the title to "How to share devices to Amazon Web Services (AWS)"?

**Comment by d0n13** [comment]
Yes. Please do. I was waiting to see what u needed to do
